test/integration/image_sharegroups_test.go

```diff
@@ -0,0 +1,237 @@
+package integration
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"testing"
+
+	"github.com/linode/linodego"
+	"github.com/stretchr/testify/require"
+)
+
+func TestImageSharing_Suite(t *testing.T) {
+	client, instance, teardown, err := setupInstance(
+		t, "fixtures/TestImageSharing_Suite", true,
+		func(client *linodego.Client, options *linodego.InstanceCreateOptions) {
+			options.Region = getRegionsWithCaps(t, client, []string{"Linodes"})[0]
+			options.Image = "linode/alpine3.22"
+		})
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(teardown)
+
+	instanceDisks, err := client.ListInstanceDisks(
+		context.Background(),
+		instance.ID,
+		nil,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// First, create a Private Image and verify that the `IsShared` and `ImageSharing` fields are as expected
+	image, err := client.CreateImage(context.Background(), linodego.ImageCreateOptions{
+		DiskID: instanceDisks[0].ID,
+		Label:  "linodego-test-image-sharing-image",
+	})
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		err := client.DeleteImage(context.Background(), image.ID)
+		require.NoError(t, err)
+	})
+
+	_, err = client.WaitForImageStatus(context.Background(), image.ID, linodego.ImageStatusAvailable, 120)
```

Comment:
The image processing time can vary dramatically by load/availability/region so I'm occasionally seeing this step timeout. Would it make sense to bump it up to `600`?